build/tasks/minify.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Minify JavaScript using SWC.
+ */
+
+"use strict";
+
+module.exports = ( grunt ) => {
+	const swc = require( "@swc/core" );
+
+	grunt.registerMultiTask(
+		"minify",
+		"Minify JavaScript using SWC",
+		async function() {
+			const done = this.async();
+			const options = this.options();
+			const sourceMapFilename = options.sourceMap && options.sourceMap.filename;
+			const sourceMapOverrides = options.sourceMap && options.sourceMap.overrides || {};
+
+			await Promise.all( this.files.map( async( { src, dest } ) => {
+				if ( src.length !== 1 ) {
+					grunt.fatal( "The minify task requires a single source per destination" );
+				}
+
+				const { code, map: incompleteMap } = await swc.minify(
+					grunt.file.read( src[ 0 ] ),
+					{
+						...options.swc,
+						inlineSourcesContent: false,
+						sourceMap: sourceMapFilename ?
+							{
+								filename: sourceMapFilename
+							} :
+							false
+					}
+				);
+
+				grunt.file.write( dest, code );
+
+				if ( sourceMapFilename ) {
+
+					// Apply map overrides if needed. See the task config description
+					// for more details.
+					const mapObject = {
+						...JSON.parse( incompleteMap ),
+						...sourceMapOverrides
+					};
+					const map = JSON.stringify( mapObject );
```

Comment:
I wish we could avoid this parse + fix + stringify back behavior, but it's still quick enough (I actually cannot measure a meaningful time difference with & without this code) and there's not much we can do until the issues are fixed or I learn which parameters to use to get what we need out of the box.